File: Qb.Net/Union.cs
using Viten.QueryBuilder.SqlOm;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Viten.QueryBuilder
{
  /// <summary>Класс реализации SQL UNION оператора</summary>
  public class Union
  {
    internal OmUnion Uni;

    internal Union()
    {
      Uni = new OmUnion();
    }

    /// <summary>Добавить объект запроса. По умолчанию UnionModifier.All</summary>
    public Union Add(Select query)
    {
      return Add(query, UnionMod.All);
    }

    /// <summary>Добавить объект запроса</summary>
    public Union Add(Select query, UnionMod modifier)
    {
      Uni.Add(Qb.GetQueryObject(query), modifier);
      return this;
    }
  }
}

File: Qb.Js/src/Union.ts
import {Select} from "./Select";
import {UnionMod} from "./Enums";
import {OmUnion} from "./OmUnion";
import {Qb} from "./Qb";
import {Utils} from "./Utils";

// interface IUnion {
//   Add(query: Select): Union;
//   Add(query: Select, modifier: UnionMode): Union;
// }
export class Union /*implements IUnion*/ {
  private Uni: OmUnion;
  constructor() {
    this.Uni = new OmUnion();
  }
  Add(query: Select): Union;
  Add(query: Select, modifier: UnionMod): Union;
  Add(): Union {
    var query: Select|undefined;
    var modifier: UnionMod = UnionMod.All;
    if (arguments.length >= 1) {
      query = arguments[0];
    }
    if (arguments.length >= 2) {
      modifier = arguments[1];
    }
    this.Uni.add(Qb.GetQueryObject(query!), modifier);
    return this;
  }
}